src/app/test/utils/getIP.ts

```diff
@@ -1,6 +1,189 @@
+import { Agent as HttpAgent } from 'node:http'
+import { Agent as HttpsAgent } from 'node:https'
 import { httpClient } from '../../../utils/httpClient'
 
-const IP_ADDRESS_PROVIDER = 'https://checkip.amazonaws.com'
-export const getIP = async () => {
-  return (await httpClient.get<string>(IP_ADDRESS_PROVIDER).then((res) => res.data)).trim()
+/**
+ * Defines the structure for network metadata configuration, including environment variables
+ * and metadata service providers (IMDS or external providers).
+ */
+interface NetworkMetadataConfig {
+  /**
+   * Predefined public IPv4 address of the server, if available.
+   * If set, no network request will be made to fetch the address.
+   */
+  publicIPv4?: string
+
+  /**
+   * Predefined public IPv6 address of the server, if available.
+   * If set, no network request will be made to fetch the address.
+   */
+  publicIPv6?: string
+
+  /**
+   * Custom IPv4 address provider endpoint (if specified by environment variables).
+   * If set, this will be queried instead of the default metadata provider.
+   */
+  ipv4Provider?: string
+
+  /**
+   * Custom IPv6 address provider endpoint (if specified by environment variables).
+   * If set, this will be queried instead of the default metadata provider.
+   */
+  ipv6Provider?: string
+
+  /**
+   * Default metadata service provider for retrieving the public IPv4 address.
+   * Can be changed to an external service.
+   */
+  defaultIPv4Provider: string
+
+  /**
+   * Default metadata service provider for retrieving the public IPv6 address.
+   * Can be changed to an external service.
+   */
+  defaultIPv6Provider: string
+
+  /**
+   * Endpoint for retrieving an authentication token required for some metadata services.
+   * Not used if an external provider is configured.
+   */
+  imdsTokenProvider: string
```

Comment:
This env variable is not used anywhere. We can just remove it tbh. This env variable depends on `X-aws-ec2-metadata-token` header anyway, I don't see any case where overriding this variable is useful

Reply:
Removed, instead using const